fiona/ogrext.pyx

```diff
@@ -335,6 +335,10 @@ cdef class OGRFeatureBuilder:
                 hh, mm, ss = value.hour, value.minute, value.second
                 OGR_F_SetFieldDateTime(
                     cogr_feature, i, 0, 0, 0, hh, mm, ss, 0)
+            elif isinstance(value, bytes) and schema_type == "bytes":
+                string_c = value
```

Comment:
@snorfalorpagus I think this is a good change, if we try to set a not-binary field to bytes we'll get a more opaque OGR error.